m4/configure_petsc.m4

```diff
@@ -63,7 +63,7 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
 #define PETSC_VERSION_GE(MAJOR, MINOR, SUBMINOR) (!PETSC_VERSION_LT(MAJOR, MINOR, SUBMINOR))
 #endif
 ]], [[
-#if ((PETSC_VERSION_GE(3, 8, 0) && PETSC_VERSION_LT(3, 13, 0)) || PETSC_VERSION_RELEASE != 1)
+#if ((PETSC_VERSION_GE(3, 16, 0) && PETSC_VERSION_LT(3, 18, 0)) || PETSC_VERSION_RELEASE != 1)
```

Comment:
I don't think the PETSc 3.17 series has started (and not sure if breaking changes will be introduced). Should we set the upper bound condition to `PETSC_VERSION_LT(3, 17, 0)`?

Reply:
Done